src/dispatch/incident/scheduled.py

```diff
@@ -222,3 +226,105 @@ def incident_close_reminder(db_session: SessionLocal, project: Project):
             # we only send the reminder for incidents that have been stable
             # longer than a week and only on Mondays
             send_incident_close_reminder(incident, db_session)
+
+
+@scheduler.add(every().monday.at("18:00"), name="incident-report-weekly")
+@timer
+@scheduled_project_task
+def incident_report_weekly(db_session: Session, project: Project):
+    """Creates and sends incident weekly reports based on notifications."""
+
+    # don't send if set to false or no notification id is set
+    if project.send_weekly_reports is False or not project.weekly_report_notification_id:
+        return
+
+    # don't send if no enabled ai plugin
+    ai_plugin = plugin_service.get_active_instance(
+        db_session=db_session, plugin_type="artificial-intelligence", project_id=project.id
+    )
+    if not ai_plugin:
+        log.warning("Incident weekly reports not sent. No AI plugin enabled.")
+        return
+
+    # we fetch all closed incidents in the last week
+    incidents = get_all_last_x_hours_by_status(
+        db_session=db_session,
+        project_id=project.id,
+        status=IncidentStatus.closed,
+        hours=24 * 7,
+    )
+
+    # no incidents closed in the last week
+    if not incidents:
+        return
+
+    storage_plugin = plugin_service.get_active_instance(
+        db_session=db_session, plugin_type="storage", project_id=project.id
+    )
+
```

Comment:
You may want to check that you have a storage_plugin here

Reply:
Good thinking. Fixed.